### PR TITLE
initial set of edge cases

### PR DESCRIPTION
**Pull request by anupsv**
## Fixes Issue

Fixes #

### CODE REVIEW

File: e2e/server_test.go [hunk 1]
```diff
@@ -6,16 +6,65 @@ import (
...
 func useMemory() bool {
 	return !runTestnetIntegrationTests
 }
 
+func isPanic(err string) bool {
+	return strings.Contains(err, "panic") && strings.Contains(err, "SIGSEGV") &&
+		strings.Contains(err, "nil pointer dereference")
+}
+
+func TestOptimismClientWithKeccak256CommitmentIntegration(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (anupsv) [reply e2e/server_test.go:26]: sure.

File: e2e/server_test.go [hunk 2]
```diff
@@ -70,6 +121,71 @@ func TestOptimismClientWithGenericCommitment(t *testing.T) {
 	require.Equal(t, testPreimage, preimage)
 }
 
+func TestProxyClientIntegrationOnly(t *testing.T) {
```
Comment (samlaf) [inline e2e/server_test.go:124]: Are these tests really testing the Client? I feel like they are testing the server instead?

**Review by samlaf — COMMENTED** [review]
Looks like useful stuff! Would be nice to have a big comment explaining the different edge cases, otherwise it's very hard to tell whether they are extensive or not.

Also should we turn this into a fuzz test using your edge cases as seed examples?

File: e2e/server_test.go [hunk 3]
```diff
@@ -6,16 +6,65 @@ import (
...
+	return strings.Contains(err, "panic") && strings.Contains(err, "SIGSEGV") &&
+		strings.Contains(err, "nil pointer dereference")
+}
+
+func TestOptimismClientWithKeccak256CommitmentIntegration(t *testing.T) {
+	if !runIntegrationTests || runTestnetIntegrationTests {
+		t.Skip("Skipping test as TESTNET env set or INTEGRATION var not set")
+	}
+
+	// nil commitment. Should return an error but currently is not. This needs to be fixed by OP
```
[the remark this answers is not in the document]
Comment (anupsv) [reply e2e/server_test.go]: done

**Comment by anupsv** [comment]
> Looks like useful stuff! Would be nice to have a big comment explaining the different edge cases, otherwise it's very hard to tell whether they are extensive or not.
> 
> Also should we turn this into a fuzz test using your edge cases as seed examples?

I've been looking to create a full end to end fuzzer, i.e integrate proxy with inabox and run the fuzzer through the proxy so the full flow is being tested continuously. Wanted to get these tests in first so that known issues are fixed. Also fuzzing, depending on seed size, would create long PR cycles but i guess people are ok with it since the holesky test takes around 40 minutes. I'll add it in a different PR later on and we can discuss there for runtimes and sizes.

File: e2e/server_test.go (hunk 2 above)
Comment (samlaf) [reply e2e/server_test.go]: Yep thanks, that works for me :)

**Comment by samlaf** [comment]
> > Looks like useful stuff! Would be nice to have a big comment explaining the different edge cases, otherwise it's very hard to tell whether they are extensive or not.
> > Also should we turn this into a fuzz test using your edge cases as seed examples?
> 
> I've been looking to create a full end to end fuzzer, i.e integrate proxy with inabox and run the fuzzer through the proxy so the full flow is being tested continuously. Wanted to get these tests in first so that known issues are fixed. Also fuzzing, depending on seed size, would create long PR cycles but i guess people are ok with it since the holesky test takes around 40 minutes. I'll add it in a different PR later on and we can discuss there for runtimes and sizes.

Agree to merging this first and then later adding fuzzing using these as base case. golang fuzzer is integrated and pretty fast! You can run it for a fixed amount of time (say 10sec), or run it until its hit n amounts of "surprising" results (say 200). Don't think it should take that long for something simple like this, but I might be wrong...

**Review by samlaf — APPROVED** [review]
